### PR TITLE
Add CreateIndex() method

### LINKED ISSUE #62: Implement CreateIndex() interface
In sync mode, need to call GetIndexState() to check whether the index has been finished.

### PR DESCRIPTION
**Pull request by ArkaprabhaChakraborty**
Fixes #62 
Signed-off-by: ArkaprabhaChakraborty <chakrabortyarkaprabha998@gmail.com>

### CODE REVIEW

**Comment by yhmo** [comment]
Still has lint error. @ArkaprabhaChakraborty

**Comment by ArkaprabhaChakraborty** [comment]
Yes Ill fix it by tomorrow.

**Comment by ArkaprabhaChakraborty** [comment]
@matrixji @yhmo how to fix the codecov error?

**Comment by matrixji** [comment]
I guess codecov is decreased due to your PR not rebasing from the latest in master. We're not setting codecov as mandatory right now.
Look's good to me, except need to squash your 3 commits to one in this PR.

**Comment by ArkaprabhaChakraborty** [comment]
Then it should have had conflicts, shouldn't it? But it doesnt say any merge conflicts or something like that.

**Comment by ArkaprabhaChakraborty** [comment]
@matrixji @yhmo

**Comment by matrixji** [comment]
Hi, @ArkaprabhaChakraborty. I see the reason why coverage failed, seems you have disabled your testing code.

**Review by matrixji — CHANGES_REQUESTED** [review]

**Comment by ArkaprabhaChakraborty** [comment]
> Hi, @ArkaprabhaChakraborty. I see the reason why coverage failed, seems you have disabled your testing code.

Yes because the current test also depends on another method which is not yet completed

**Comment by ArkaprabhaChakraborty** [comment]
I want to make changes to this test later as it requires DescribeIndex() method too

File: test/it/TestCreateIndex.cpp [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    params["nlevel"] = "10";
+
+    milvus::IndexDesc index_desc(field_name, index_name, index_id, params);
+    const auto progress_monitor = ::milvus::ProgressMonitor::NoWait();
+
+    milvus::proto::milvus::CreateIndexRequest request;
+    request.set_collection_name(collection_name);
+    request.set_db_name(index_name);
+    request.set_field_name(field_name);
+
```
[the remark this answers is not in the document]
Comment (ArkaprabhaChakraborty) [reply test/it/TestCreateIndex.cpp:33]: I don't know how to add this.

File: test/it/TestCreateIndex.cpp [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+#include "mocks/MilvusMockedTest.h"
+
+using ::milvus::StatusCode;
+using ::milvus::proto::milvus::CreateIndexRequest;
+using ::testing::_;
+using ::testing::AllOf;
+using ::testing::Property;
+
+TEST_F(MilvusMockedTest, DISABLED_TestCreateIndex) {
```
[the remark this answers is not in the document]
Comment (ArkaprabhaChakraborty) [reply test/it/TestCreateIndex.cpp:11]: Yes I have DISABLED the test as it requires DescribeIndex() too. But the error was:
```
Uninteresting mock function call - returning default value.
Function call: CreateIndex(0x7f35dc00c200, 0x7f35dc00ae50, 0x7f35f144e790)
          Returns: 72-byte object <00-00 00-00 00-00 00-00 30-CA 00-DC 35-7F 00-00 00-00 00-00 00-00 00-00 00-6B 12-F5 35-7F 00-00 50-1B 00-DC 35-7F 00-00 50-CA 00-DC 35-7F 00-00 00-00 00-00 00-00 00-00 00-56 58-86 3A-69 ED-60 C0-E3 44-F1 35-7F 00-00>
```

**Comment by matrixji** [comment]
> I want to make changes to this test later as it requires DescribeIndex() method too

For now, as we are in the early stage, IMO it's acceptable to add test code later.
/lgtm

**Pull request merged**